### ISSUE TITLE
RTMP failing in Firefox

### ISSUE THREAD

**Issue by Tralapo** [videojs/video.js#1994]
Hello, I'm using VideoJS for livestreams with rtmp and hls. My setup:

```html
<video 
id="Video" 
class="video-js vjs-default-skin vjs-big-play-centered" 
controls 
preload="none" 
width="auto" 
height="auto" 
poster="poster.jpg" 
data-setup='{"techOrder": ["flash", "html5", "other supported tech"], "nativeControlsForTouch": true, "controlBar": { "muteToggle": false, "volumeControl": false, "timeDivider": false, "durationDisplay": false, "progressControl": false } }'
>
<source src="rtmp://url.ltd/live.stream" type='rtmp/mp4'/>
<source src="http://url.ltd/live.stream/playlist.m3u8" type='video/mp4'/></video>
```

This works fine in Chrome, IE, Opera, on iOS and on Android etcetera.
But it's not working in Firefox. Not on desktop and not on Android. I get this error:

```The video could not be loaded, either because the server or network failed or because the format is not supported.```

In the Firefox developer console, I can see it's doing it wrong:

```
No suitable players found and fallback enabled
HTTP “Content-Type” from “application/vnd.apple.mpegurl” is not supported. Load of media resource http://url.ltd/live.stream/playlist.m3u8 failed.
```
So, if I understand this well, it's trying to load the HLS video. Why? It should just load the RTMP-version in the flash playback, like all the other browsers.

Searching Google I saw some people had slightly similar problems and they suggest adding this to my .htaccess:

```
AddType video/webm .webm
AddType video/ogg .ogv
AddType video/mp4 .mp4
```
That isn't working for me though. What goes wrong here?

*Edit, after another load, I see Firefox does try the RTMP first, console now looks like this:
```
Specified “type”-attribute “rtmp/mp4” is not supported. Load of media resource rtmp://url.ltd/live.stream failed.
No suitable players found and fallback enabled
HTTP “Content-Type” from “application/vnd.apple.mpegurl” is not supported. Load of media resource http://url.ltd/live.stream/playlist.m3u8 failed.
```

**Comment by gkatsev** [comment]
You are the wrong mimetype set up for the m3u8 playlist. You want to use `application/vnd.apple.mpegurl` or `application/x-mpegurl`. Videojs ends up trying to use that source because it thinks it can play it, but it actually can't play it. Videojs doesn't really retry other sources once it chooses a source.

**Comment by Tralapo** [comment]
I find it strange you close this issue this fast, without waiting for my response.

Because the solution you give, doesn't work. Both of them. Now I get this error:

```
"VIDEOJS:" "ERROR:" "The "Other supported tech" tech is undefined. Skipped browser support check for that tech."
"VIDEOJS:" "ERROR:" "(CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED)" "No compatible source was found for this video." Object { code: 4, message: "No compatible source was found for this video." }
```

**Comment by mmcc** [comment]
@Tralapo don't worry, everyone still gets the email updates, so other people can still chime in. We're always willing to help out, but this is an issue tracker, not a support forum, so we try to keep open issues limited to confirmed bugs and things of that nature with the core player.

TL;DR: It sounds like something is wrong with Flash in your Firefox installation.

That being said, you should read those error messages a little more closely. "other supported tech" isn't a tech, I assume it was just an example of how you could list other techs in a README somewhere. I don't think this is an issue here, but you've still got a typo in your HLS mime type (should be x-mpegur**l**).

At the root of this, I'm not sure what you mean by "fallback" in this case. You're specifying that Flash should be the first source used, which is what we generally consider a fallback, so perhaps you're referring to the m3u8. HLS is only natively supported by Safari, so Firefox isn't going to be able to play that anyway, which means it should be using the RTMP source with Flash. If that's not working, but it's working in Chrome, my guess is something is wrong with Flash in your FF installation.

**Comment by Tralapo** [comment]
@mmcc Oh God. "other supported tech" is how it's showed in your own docs. I never tought that was just text, I tought it had a function.

About the ```application/vnd.apple.mpegurl``` or ```application/x-mpegurl```. Android and iOS do play the m3u8 without any trouble when called ```video/mp4``` so that doesn't seem to be a problem.

I call Flash with RTMP the fallback, I'm sorry if I wasn't clear about it. I now deleted Firefox and installed it all over again, and i now seems to work again. A bit strange, because I had several complaints from visitors, so I would think something went wrong with an update mechanism in Firefox or something? Good it's fixed now. Thanks.

And for my information: Is there a support forum? I couldn't find it, that's why I end op here.

**Comment by mmcc** [comment]
We should probably make that example a little more explicit, but it is an important thing to note because there are third party techs such as HLS that could be listed in that techOrder.

We generally suggest using StackOverflow and use the [Video.js tag](https://stackoverflow.com/questions/tagged/video.js). We tried to run our own support forum for a while, but the built in community around SO means *much* better chances of questions getting answered in a reasonable time.
